File: tests/uasm/sample/array_index_mutation/ArrayIndexMutationTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class ArrayIndexMutationTest : UdonSharpBehaviour
{
    public void Start()
    {
        int[] values = new int[] { 1, 2, 3, 4 };
        values[1] = values[0] + values[2];
        values[3] = values[1] * 2;

        Debug.Log(values[1]);
        Debug.Log(values[3]);
    }
}

File: tests/uasm/sample/array_index_mutation/ArrayIndexMutationTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class ArrayIndexMutationTest extends UdonSharpBehaviour {
  Start(): void {
    const values: number[] = [1, 2, 3, 4];
    values[1] = values[0] + values[2];
    values[3] = values[1] * 2;

    Debug.Log(values[1]);
    Debug.Log(values[3]);
  }
}
